docs/DEVGUIDE.md

```diff
@@ -12,6 +12,7 @@ layout:
     ├── build          # Contains build-related scripts and subdirectories containing Dockerfiles
     ├── cmd            # Contains "main" Go packages for each service catalog component binary
     ├── contrib        # Contains all non-essential source
+    │   └── hack       # Non-build related scripts
```

Comment:
I think including the reason for each grouping (ie dirs) is a good idea if we're trying to explain why the groupings are there